cmake/CMakeLists.txt

```diff
@@ -127,11 +127,12 @@ option(onnxruntime_DISABLE_ORT_FORMAT_LOAD "Disable loading an ORT format model
 option(onnxruntime_ENABLE_NVTX_PROFILE "Enable NVTX profile." OFF)
 option(onnxruntime_ENABLE_TRAINING "Enable training functionality." OFF)
 option(onnxruntime_ENABLE_TRAINING_E2E_TESTS "Enable training end-to-end tests." OFF)
-option(onnxruntime_USE_HOROVOD "Build with HOROVOD support" OFF)
+option(onnxruntime_ENABLE_ADVANCED_INSTRUCTIONS_FOR_TRAINING "Build with advanced instruction sets" OFF)
```

Comment:
Sure, what about something like ENABLE_CPU_FP16_OPS? 
I put it as advanced ISA since some of the instruction subsets (f16c/fma) are only available after Ivybridge. Another reason for defaulting this to OFF is that they are not required to run inferencing or training